src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/job/CompetencyExtractionJob.java

```diff
@@ -3,16 +3,17 @@
 import com.fasterxml.jackson.annotation.JsonInclude;
 
 import de.tum.cit.aet.artemis.core.domain.Course;
+import de.tum.cit.aet.artemis.core.domain.User;
 
 /**
  * A pyris job that extracts competencies from a course description.
  *
- * @param jobId     the job id
- * @param courseId  the course in which the competencies are being extracted
- * @param userLogin the user login of the user who started the job
+ * @param jobId    the job id
+ * @param courseId the course in which the competencies are being extracted
+ * @param user     the user who started the job
  */
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public record CompetencyExtractionJob(String jobId, long courseId, String userLogin) implements PyrisJob {
+public record CompetencyExtractionJob(String jobId, long courseId, User user) implements PyrisJob {
```

Comment:
Maybe I'm mistaken, but this code smells a little bit: I'm not sure it's a good idea to store a whole `User` object in Hazelcast. This is meant to be a lightweight cache object, which is why all the `PyrisJob` records only contained IDs until now. @Hialus, what do you think?